### PR TITLE
DIMS2.1.2 release

### PR DESCRIPTION
**Pull request by FiniDG**
bug fix and new possibilities for positive control sample names
https://github.com/UMCUGenetics/DIMS/pull/45
https://github.com/UMCUGenetics/DIMS/pull/46

DIMS v2.1.2 is an update of the DIMS pipeline to make other positive control names possible. At this point, they are hard coded as P1002.1, P1003.1 and P1005.1. Recently there is an update of patient materials and therefore plasma runs will use P1005.2 and DBS runs still makes use of P1005.1. 

Bug with DIMS2.1.1. Samplenames with only one number (e.g. P7.2, C3.1) are blocked by the GUI as "incorrect sample names that do not fit the requirements). This was a small oversight in the regex, which expected 2 numbers. Those sample names are not incorrect and now the GUI will not block these anymore.

fixes issue #47

### CODE REVIEW

File: pipeline/scripts/13-excelExport.R [hunk 1]
```diff
@@ -398,39 +398,54 @@ ggsave(paste0(outdir, "/plots/IS_line_select_sum.png"), plot = IS_sum_selection_
...
-    if (!(pos %in% colnames(outlist))) {
-      missing_pos <- c(missing_pos, pos)
-    }}
+  # any() grep because you get a vector of FALSE's and TRUE's. only one grep match is needed for each positive control
+  if (any(grep("^(P1002\\.)[[:digit:]]+_", positivecontrol_list)) && 
+      any(grep("^(P1003\\.)[[:digit:]]+_", positivecontrol_list)) && 
+      any(grep("^(P1005\\.)[[:digit:]]+_", positivecontrol_list))){
+    cat("All three positive controls are present")
+  } else {
+    missing_pos <- paste0(c("positive controls list is not complete: ", positivecontrol_list), collapse=" ")
```
Comment (iljavh) [inline pipeline/scripts/13-excelExport.R:418]: This comment should be clearer, as the user could interpret it the other way around. As if these are the missing pos controls.

Comment (iljavh) [reply pipeline/scripts/13-excelExport.R:418]: Naming should be more intuitive, it used to be the missing pos contrs, but now it's the opposite of missing. Change the name of the variable, warning text, and file name in line 458.

Comment (FiniDG) [reply pipeline/scripts/13-excelExport.R]: changed name of the missing_pos variable and text in e-mail to be more clear

**Review by iljavh — APPROVED** [review]
Changes look good

**Pull request merged**